frontend/www/js/omegaup/arena/contest_practice.ts

```diff
@@ -21,43 +27,171 @@ OmegaUp.on('ready', () => {
     data: () => ({
       problemInfo: null as types.ProblemInfo | null,
       problem: null as ActiveProblem | null,
+      problems: payload.problems as types.NavbarProblemsetProblem[],
       clarifications: payload.clarifications,
     }),
+    methods: {
+      getMaxScore: (
+        runs: types.Run[],
+        alias: string,
+        previousScore: number,
+      ): number => {
+        let maxScore = previousScore;
+        for (const run of runs) {
+          if (alias != run.alias) {
+            continue;
+          }
+          const score = run.contest_score || 0;
+          if (score > maxScore) {
+            maxScore = score;
+          }
+        }
+        return maxScore;
+      },
+    },
     render: function (createElement) {
       return createElement('omegaup-arena-contest-practice', {
         props: {
           contest: payload.contest,
           contestAdmin: payload.contestAdmin,
-          problems: payload.problems,
+          problems: this.problems,
           users: payload.users,
           problemInfo: this.problemInfo,
           problem: this.problem,
           clarifications: this.clarifications,
           activeTab,
         },
         on: {
-          'navigate-to-problem': (source: ActiveProblem) => {
+          'navigate-to-problem': (request: ActiveProblem) => {
             api.Problem.details({
-              contest_alias: payload.contest.alias,
-              problem_alias: source.alias,
+              problem_alias: request.problem.alias,
               prevent_problemset_open: false,
             })
               .then((problemInfo) => {
+                for (const run of problemInfo.runs ?? []) {
+                  trackRun(run);
+                }
                 const currentProblem = payload.problems?.find(
                   ({ alias }) => alias == problemInfo.alias,
                 );
                 problemInfo.title = currentProblem?.text ?? '';
                 contestPractice.problemInfo = problemInfo;
-                source.alias = problemInfo.alias;
-                source.runs = problemInfo.runs ?? [];
-                window.location.hash = `#problems/${source.alias}`;
+                request.problem.alias = problemInfo.alias;
+                request.runs = myRunsStore.state.runs;
+                (request.problem.bestScore = this.getMaxScore(
+                  request.runs,
+                  problemInfo.alias,
+                  0,
+                )),
+                  (window.location.hash = `#problems/${request.problem.alias}`);
```

Comment:
```suggestion
                request.problem.bestScore = this.getMaxScore(
                  request.runs,
                  problemInfo.alias,
                  0,
                );
                window.location.hash = `#problems/${request.problem.alias}`;
```

el uso del [operador coma](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Comma_Operator) es intencional? se ve rarísimo, y es mejor dejarlos como statements independientes.

Reply:
No sé si fue el linter, porque no había usado ese operador para algo intencional. Lo dejo así como indicas